src/zenml/pipelines/base_pipeline.py

```diff
@@ -756,3 +752,242 @@ def write_run_configuration_template(
+                name=step_spec.pipeline_parameter_name,
+                kind=inspect.Parameter.POSITIONAL_OR_KEYWORD,
+            )
+            for step_spec in model.spec.steps
+        ]
+        signature = inspect.Signature(parameters=parameters)
+        connect.__signature__ = signature  # type: ignore[attr-defined]
+
+        return connect
+
+    def register(self) -> "PipelineResponseModel":
+        """Register the pipeline in the server.
+
+        Returns:
+            The registered pipeline model.
+        """
+        pipeline_spec = Compiler().compile_spec(self)
+        return self._register(pipeline_spec=pipeline_spec)
+
+    def _register(
+        self, pipeline_spec: "PipelineSpec"
+    ) -> "PipelineResponseModel":
+        """Register the pipeline in the server.
+
+        Args:
+            pipeline_spec: The pipeline spec to register.
+
+        Returns:
+            The registered pipeline model.
+        """
+        version_hash = self._compute_unique_identifier(
+            pipeline_spec=pipeline_spec
+        )
+
+        client = Client()
+        matching_pipelines = client.list_pipelines(
+            name=self.name,
+            version_hash=version_hash,
+            size=1,
+            sort_by="desc:created",
+        )
+        if matching_pipelines.total:
+            registered_pipeline = matching_pipelines.items[0]
+            logger.info(
+                "Reusing registered pipeline `%s` (version: %s).",
+                registered_pipeline.name,
+                registered_pipeline.version,
+            )
+            return registered_pipeline
+
+        latest_version = self._get_latest_version() or 0
+        version = str(latest_version + 1)
+
+        request = PipelineRequestModel(
+            workspace=client.active_workspace.id,
+            user=client.active_user.id,
+            name=self.name,
+            version=version,
+            version_hash=version_hash,
+            spec=pipeline_spec,
+            docstring=self.__doc__,
+        )
+
+        registered_pipeline = client.zen_store.create_pipeline(
+            pipeline=request
+        )
+        logger.info(
+            "Registered pipeline `%s` (version %s).",
+            registered_pipeline.name,
+            registered_pipeline.version,
+        )
+        return registered_pipeline
+
+    def _compute_unique_identifier(self, pipeline_spec: PipelineSpec) -> str:
+        """Computes a unique identifier from the pipeline spec and steps.
+
+        Args:
+            pipeline_spec: Compiled spec of the pipeline.
+
+        Returns:
+            The unique identifier of the pipeline.
+        """
+        hash_ = hashlib.md5()
+        hash_.update(pipeline_spec.json(sort_keys=False).encode())
+
+        for step_spec in pipeline_spec.steps:
+            step_source = self.steps[
+                step_spec.pipeline_parameter_name
+            ].source_code
+            hash_.update(step_source.encode())
```

Comment:
Not sure about this, but should the step source become part of the pipeline spec perhaps?

Reply:
I don't think so, the spec should not contain any source code (which isn't even enough to run the step, so it really would just be useless anyway)